src/backend/distributed/utils/metadata_cache.c

```diff
@@ -2636,7 +2658,7 @@ InitializeDistTableCache(void)
 
 	fmgr_info_cxt(F_OIDEQ,
 				  &DistShardScanKey[0].sk_func,
-				  CacheMemoryContext);
+				  shardCacheMemoryContext);
```

Comment:
To make deleting the memory context effective, we should probably also use shardCacheMemoryContext for the hash tables (currently they use `TopMemoryContext`, which is the default). Now it only allocates these two functions in the memory context.

Reply:
done